### PR TITLE
[ADD] 'pos_product_template'

### PR DESCRIPTION
**Pull request by legalsylvain**
Propose a module to select variant of template.

Main features:
- products variant are now hidden from the POS (only one template is displayed);
- it is still possible to scan variant;
- an extra screen is displayed to select variant when a template is selected;

see https://www.youtube.com/watch?v=sCZr39NxSlU&feature=youtu.be

Review and test are welcome.

### CODE REVIEW

**Comment by sebastienbeau** [comment]
:+1: Great job Sylvain, I test it and it's work perfectly ! Thanks

File: pos_product_template/static/src/css/ppt.css [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    background: none repeat scroll 0% 0% #7F82AC;
+    padding: 2px 5px;
+    border-radius: 2px;
+    font-size: 9px;
+    margin-right:3px;
+}
+
+
+/*  ********* Variant Display  ********************************************* */
+/* This part is a copy past from Odoo Project, with some values changed      */
```
Comment (sylvainc) [inline pos_product_template/static/src/css/ppt.css]: typo: s/past/paste
What about using mulitple classes (class="product variant") in the QWeb template and ".pos .product.variant" in the CSS, and just redefine the attributes you need ?

File: pos_product_template/static/src/js/ppt.js [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+Define : pos_product_template.SelectVariantPopupWidget
+    
+- This widget that display a pop up to select a variant of a Template;
+*********************************************************** */
+    module.SelectVariantPopupWidget = module.PopUpWidget.extend({
+        template:'SelectVariantPopupWidget',
+
+        init: function(parent, options) {
+            this._super(parent,options);
+            this.template_name = '';
```
Comment (sylvainc) [inline pos_product_template/static/src/js/ppt.js]: this.template_name seems unsued

File: pos_product_template/static/src/js/ppt.js [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+                self.hide();
+            });
+        },
+
+        show: function(product_tmpl_id){
+            var self = this;
+            var template = this.pos.db.template_by_id[product_tmpl_id];
+
+            // Display Name of Template
+            this.template_name = template.name;
```
Comment (sylvainc) [inline pos_product_template/static/src/js/ppt.js]: same here: this.template_name seems unsued

File: pos_product_template/static/src/js/ppt.js [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+    module.AttributeListWidget = module.PosBaseWidget.extend({
+        template:'AttributeListWidget',
+
+        init: function(parent, options) {
+            var self = this;
+            this.attribute_list = [];
+            this.product_template = null;
+            this.click_set_attribute_handler = function(event){
+                /*TODO: Refactor this function with elegant DOM manipulation */
+                // remove selected item
```
Comment (sylvainc) [inline pos_product_template/static/src/js/ppt.js:285]: Maybe using JQuery could help

**Comment by sylvainc** [comment]
Well done!

**Comment by AdilHoumadi** [comment]
Very nice, good job :+1:

**Comment by pedrobaeza** [comment]
Hi, Sylvain, great module!

Can you please provide pot translation file? Addressing this and @sylvainc's issues, I will merge it.

File: pos_product_template/static/src/js/ppt.js (hunk 4 above)
Comment (legalsylvain) [inline pos_product_template/static/src/js/ppt.js:285]: Yes, I tried, but I didn't manage with JQuery. But I'm not JQuery Expert. Maybe could we accept this PR like that, as it's not a bug but only a 'not very clear code'. What do you think about that point ?

Comment (sylvainc) [inline pos_product_template/static/src/js/ppt.js:285]: Hi, it's ok for me.
Thanks!

**Comment by legalsylvain** [comment]
Hi all, 
i just 
- Fixed a bug that @sebastienbeau said me: There was a bug that displayed "value : 0 products" if there was no product with this value for the current template;See extra-demo data added: All the IPad has the same value "2.4Ghz" for the attribute "Wifi"; So display other values are not relevant;
- added some extra-description (comment and screenshot of the module);

Regards.

**Comment by pedrobaeza** [comment]
Great, so I proceed with the merge and if any other improvement is found, we make another PR.

**Pull request merged**